.circleci/workflows.yml

```diff
@@ -122,9 +122,7 @@ jobs:
             # Remove other versions so they don't end up in the cache.
             for x in $(ls -1 ${NVM_DIR}/versions/node | grep -v $(nvm version default)); do nvm uninstall $x; done
       - run: npm install yarn --location=global
```

Comment:
Because yarn is yet another package manager, same like npm. 
It is preferred that it is installed globally because otherwise it would not make much sense to run it via another package manager like: `npx yarn ...`

Reply:
For that question, i believe we need to loop in @manuelwedler and get his input on why yarn was used in the first place. I have no strong opinion on using strictly yarn or npm, i am up for switching to npm as well.

Reply:
First of all, we had some bad experiences in Raiden with npm and yarn was always reliable. For example, I had flaky installations with npm. That's why we started to use yarn basically everywhere. 

I don't completely get what should be achieved here but I have a lot of questions..
- Why is yarn installed and then, npm is used for installing the dependencies (in this step)? Not sure in which directory this is run, but as far as I see we only have yarn.lock files (and not the npm lock files) in our repo. So we should also use yarn in the CI.
- Why don't we use an image that has node preinstalled? They usually come with the package managers installed already I think.
- We have this circleci/node orb declared on the top. Do I get it correctly that we only use it for the frontend workflow?

One important hint: Always use `yarn install --frozen-lockfile` or the corresponding `npm ci` for installing packages in CI environments. The mentioned orb does that automatically.

Reply:
> I don't completely get what should be achieved here but I have a lot of questions..

:-)

> Why is yarn installed and then, npm is used for installing the dependencies (in this step)? Not sure in which directory this is run, but as far as I see we only have yarn.lock files (and not the npm lock files) in our repo. So we should also use yarn in the CI.

I think the reason was purely historical as Paul came up with the original version of the relayer and I didn't really know to mess too much with it due to my lack of knowledge of the node ecosystem and best practices.
If yarn is the preferred way to install packages, then, let's switch to it, by all means.  :+1: 

> Why don't we use an image that has node preinstalled? They usually come with the package managers installed already I think.

Eh... this is where it gets a bit complicated. We can't use simple Docker containers for our tests because the networking setup will not work (long story). So we had to go with a VM. But in that VM we need both Python and node packages, which is the reason we cache the entire node runtime + packages.

> We have this circleci/node orb declared on the top. Do I get it correctly that we only use it for the frontend workflow?

Yes. The frontend is completely separate and there is no need to run that workflow in a VM, which itself incurs ~25s overhead, compared to a <1s for a Docker container.

Reply:
I'm all for just using yarn. You need to run the installation locally with `yarn install` to generate a `yarn.lock` file and replace the `package-lock.json` here with it.

Reply:
✅